src/mapping/PartitionOfUnityMapping.hpp

```diff
@@ -0,0 +1,385 @@
+    setOutputRequirement(Mapping::MeshRequirement::FULL);
+  } else {
+    setInputRequirement(Mapping::MeshRequirement::VERTEX);
+    setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+  }
+  // Simply set each axis to be active according to the space dimension.
+  // We could even set each axis to be true, which would just lead to 'zero' columns in the polynomial QR.
+  // Integrated polynomials are anyway not supported with PoU
+  _deadAxis.clear();
+  std::fill_n(std::back_inserter(_deadAxis), getDimensions(), false);
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void PartitionOfUnityMapping<RADIAL_BASIS_FUNCTION_T>::tagMeshFirstRound()
+{
+  PRECICE_TRACE();
+  mesh::PtrMesh filterMesh, outMesh;
+  if (this->hasConstraint(Mapping::CONSERVATIVE)) {
+    filterMesh = this->output(); // remote
+    outMesh    = this->input();  // local
+  } else {
+    filterMesh = this->input();  // remote
+    outMesh    = this->output(); // local
+  }
+
+  if (outMesh->vertices().empty())
+    return; // Ranks not at the interface should never hold interface vertices
+
+  // TODO: Check again the tagging in combination with the partition construction (which mesh to use)
+  // In order to construct the local partitions, we need all vertices with a distance of 2 x radius,
+  // as the relevant partitions centers have a maximum distance of radius, and the proper construction of the
+  // interpolant requires all vertices with a distance of radius from the center.
+  auto bb = outMesh->getBoundingBox();
+
+  if (averagePartitionRadius == 0)
+    averagePartitionRadius = partitioner::estimatePartitionRadius(_verticesPerPartition, filterMesh, bb);
+
+  // @TODO: This assert is not completely right, as it checks all dimensions for non-emptyness (which might not be the case).
+  // However, with the current BB implementation, the expandBy function will just do nothing.
+  PRECICE_ASSERT(!bb.empty());
+  PRECICE_ASSERT(averagePartitionRadius > 0);
+  // Now we extend the bounding box by the radius
+  bb.expandBy(1 * averagePartitionRadius);
+
+  // ... and tag all affected vertices
+  auto verticesNew = filterMesh->index().getVerticesInsideBox(bb);
+
+  std::for_each(verticesNew.begin(), verticesNew.end(), [&filterMesh](VertexID v) { filterMesh->vertices()[v].tag(); });
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void PartitionOfUnityMapping<RADIAL_BASIS_FUNCTION_T>::tagMeshSecondRound()
+{
+  // Probably nothing to be done here. There is no global ownership for matrix entries required and we tag all potentially locally relevant vertices already in the first round.
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void PartitionOfUnityMapping<RADIAL_BASIS_FUNCTION_T>::computeMapping()
+{
+  PRECICE_TRACE();
+
+  precice::utils::Event e("map.pou.computeMapping.From" + this->input()->getName() + "To" + this->output()->getName(), precice::syncMode);
+
+  // Recompute the whole partitioning
+  this->clear();
```

Comment:
Feels a bit odd. Normally, I would expect a `assertion(not hasComputedMapping())` or similar here and do the "clearing" somewhere outside.

Reply:
I don't really understand this comment. Why is it odd? We first have to clear the vector containing the clusters, because `compute_mapping` uses `emplace_back` in order to store the clusters in a vector. Calling `compute_mapping` here multiple times would result in erroneous behavior. In usual use-cases, this should never happen, but we use it a lot for the unit testing.

Reply:
> In usual use-cases, this should never happen, but we use it a lot for the unit testing.

But we want to run into such a problem (if we ever have it), right? So `assertion(not hasComputedMapping())` would be helpful, no?

Reply:
Added the `clear` to the test